contracts/extensions/EIP712.sol

```diff
@@ -1,18 +1,19 @@
 // SPDX-License-Identifier: UNLICENSED
```

Comment:
instead of adding this file to the repo, I think it's better to use 
```
import "@openzeppelin/contracts-upgradeable/utils/cryptography/EIP712Upgradeable.sol";
```